File: java/code-duplication/good/TaxCalculator.java
class TaxCalculator {
    private static final java.util.Map<String, Double> TAX_RATES = java.util.Map.of(
        "CA", 0.0825,
        "NY", 0.04,
        "TX", 0.0625
    );

    public double calculateTax(double price, String state) {
        double taxRate = TAX_RATES.getOrDefault(state, 0.0);
        return price * taxRate;
    }
}

class ShippingCalculator {
    private static final double BASE_RATE = 5.0;
    private static final double WEIGHT_RATE_PER_UNIT = 0.5;
    private static final double DISTANCE_RATE_PER_UNIT = 0.1;

    public double calculateShipping(double weight, double distance) {
        double weightRate = weight * WEIGHT_RATE_PER_UNIT;
        double distanceRate = distance * DISTANCE_RATE_PER_UNIT;

        return BASE_RATE + weightRate + distanceRate;
    }
}

File: python/code-duplication/good/tax_calculator.py
class TaxCalculator:
    TAX_RATES = {
        'CA': 0.0825,
        'NY': 0.04,
        'TX': 0.0625,
    }

    def calculate_tax(self, price: float, state: str) -> float:
        tax_rate = self.TAX_RATES.get(state, 0.0)
        return price * tax_rate


class ShippingCalculator:
    BASE_RATE = 5.0
    WEIGHT_RATE_PER_UNIT = 0.5
    DISTANCE_RATE_PER_UNIT = 0.1

    def calculate_shipping(self, weight: float, distance: float) -> float:
        weight_rate = weight * self.WEIGHT_RATE_PER_UNIT
        distance_rate = distance * self.DISTANCE_RATE_PER_UNIT

        return self.BASE_RATE + weight_rate + distance_rate
